code/modules/reagents/chemistry/reagents/blob_reagents.dm

```diff
@@ -199,11 +199,9 @@
 		M.reagents.add_reagent("spore", 0.2*reac_volume)
 	M.apply_damage(0.7*reac_volume, TOX)
 
-/datum/reagent/blob/regenerative_materia/on_mob_life(mob/living/M)
-	M.adjustToxLoss(1*REM)
-	if(iscarbon(M))
-		var/mob/living/carbon/N = M
-		N.hal_screwyhud = SCREWYHUD_HEALTHY //fully healed, honest
+/datum/reagent/blob/regenerative_materia/on_mob_life(mob/living/carbon/C)
+	C.adjustToxLoss(1*REM)
+	C.hal_screwyhud = SCREWYHUD_HEALTHY //fully healed, honest
```

Comment:
Does this actually do anything for non-humans ? Not really relevant just making sure it doesn't break updates completely.

Reply:
no idea, but it's the same code, it has always run for everyone capable of getting this proc called on them